File: exercises/codejam/2010_africa/qualification/store_credit/credit.clj
(defn to-int [n] 
  (Integer/parseInt n))

(def n (to-int (read-line)))

(defn items-to-buy [credit items]
  (for [i (range (count items))
        j (range (count items))
        :let [item-i (nth items i)
              item-j (nth items j)]
        :when (and (not= i j)
                   (= credit (+ item-i item-j)))]
    [(inc i) (inc j)]))

(doseq [i (range n)
        :let [credit (to-int (read-line))
              items-size (read-line)
              items (map to-int (clojure.string/split (read-line) #" "))]]
  (println (str "Case #" (inc i) ": " (clojure.string/join " " (first (items-to-buy credit items))))))

File: exercises/codejam/2010_africa/qualification/store_credit/credit.rb

def items_to_buy(credit, items)
  items.each_with_index do |item1, i|
    items.each_with_index do |item2, j|
      next if i == j
      
      return [i+1, j+1] if credit == item1 + item2
    end
  end
end


n = gets.to_i

n.times do |i|
  credit = gets.to_i  
  gets
  items = gets.split.map { |n| n.to_i }

  puts "Case \##{i+1}: #{items_to_buy(credit, items).join(' ')}"
end
